commons/src/main/java/io/aiven/kafka/tieredstorage/commons/UniversalRemoteStorageManagerConfig.java

```diff
@@ -37,6 +38,20 @@ public class UniversalRemoteStorageManagerConfig extends AbstractConfig {
     private static final String OBJECT_STORAGE_KEY_PREFIX_CONFIG = "key.prefix";
     private static final String OBJECT_STORAGE_KEY_PREFIX_DOC = "The object storage path prefix";
 
+    private static final String SEGMENT_MANIFEST_CACHE_PREFIX = "segment.manifest.cache.";
+    private static final String SEGMENT_MANIFEST_CACHE_SIZE_CONFIG = SEGMENT_MANIFEST_CACHE_PREFIX + "size";
+    private static final Long SEGMENT_MANIFEST_CACHE_SIZE_DEFAULT = 1000L;  // TODO consider a better default
```

Comment:
I thought about that. Technically can be done with weights as you propose.
My thinking is that it may be more convenient from the operation point of view to specify the number of elements instead of bytes. But may be vice versa, depending on the situation. Ideally we should have a possibility for both limits. How do you feel if we make a ticket to address this separately?

Reply:
We can create a separate ticket foe this, yes. But IMO from operational perspective count does not say anything unless you know the chunk size(that can be non constant in case we decide to cache compressed chunks). On the other hand if you can estimate the size of the cache then you can guess appropriate limits plus expose the size as a metric AFAIU. And having such a metric we can create a CLI tool to free up the memory consumed by cache in case of any unexpected situations.

Reply:
I see your point and it's valid. I will replace the item size limit with weight

Reply:
I managed to calculate object sizes in the run time with [JOL](https://github.com/openjdk/jol). However, I don't feel much like bringing this library as a dependency. It's pretty hacky, uses `Unsafe` and stuff like that and it was probably never meant to run in production.
So, we'll have to live with the item count limit. Considering, we're not speaking about using the last bits of memory for this cache (it should be probably single digit megabytes), this should be fine. We will run some estimate and add to the README how much a typical item will weigh on popular setups.

Reply:
Discussed more offline and I agree, at least for now count is fine.